sympy/functions/elementary/trigonometric.py

```diff
@@ -845,6 +849,10 @@ def _eval_rewrite_as_sec(self, arg, **kwargs):
     def _eval_rewrite_as_csc(self, arg, **kwargs):
         return 1/sec(arg).rewrite(csc, **kwargs)
 
+    def _eval_rewrite_as_besselj(self, arg, **kwargs):
+        from sympy.functions.special.bessel import besselj
+        return Piecewise(((sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg)/2), Ne(arg, 0)), (1, True))
```

Comment:
I have said several times already that long lines should be wrapped. If you just look at the code in the rest of the file you can clearly see that the lines you have added stick out as being much longer.

I would write this as
```python
        return Piecewise(
            ((sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg)/2), Ne(arg, 0)),
            (1, True)
        )
```
That way it is visually clear what the different cases in the Piecewise are.

Also the `2` and the `sqrt(2)` can be cancelled (as mentioned before).